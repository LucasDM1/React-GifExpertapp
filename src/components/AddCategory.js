import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

    const [inputValue, setinputValue] = useState("");

    const handleInputChange= (e) =>{

        setinputValue(e.target.value);

        console.log("handleInput ejecutado" )
    }

    const handleSubmit = (e) =>{
        e.preventDefault();
        console.log("submiteado")
        if (inputValue.trim().length>2) {
            setCategories(catg => [ inputValue, ...catg])
            setinputValue('')
        }

       
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input 
            type="text"
            value={inputValue}
            onChange={(e)=>{handleInputChange(e)}}
            />
        </form>
    )

   
}
 AddCategory.protoTypes = {
        setCategories: PropTypes.func.isRequired
    };