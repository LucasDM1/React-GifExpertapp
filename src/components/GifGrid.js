import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
//import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

    // const [images, setImages] = useState([]);

    // useEffect(() => {
    //     getGifs(category)
    //         .then(setImages)
    // }, [category])
    const {loading, data} = useFetchGifs(category);
    

    return (
        <>
        <h2 className="animate__animated animate__fadeIn">{category}</h2>
        {loading && <p>Loading... </p>}
       
        <div className="card-grid">
            

            {data.map( img => (
                <GifGridItem 
                key={img.id}
                { ...img } 
                //img={img}
                />
            ))}
           
        </div>
        </>
    )
}

GifGrid.propTypes={
    category: PropTypes.string.isRequired

}