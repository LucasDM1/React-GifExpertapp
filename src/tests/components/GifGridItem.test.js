import React from 'react';
import { shallow } from 'enzyme';

import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas de <GifGridItem />', () => {

    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow( <GifGridItem title={title} url={url} />)

    test('Debe mostrar GifGridItem correctamente', () => {
        expect(wrapper).toMatchSnapshot()
        
    })
    
    test('Debe tener un h3 con el texto ', () => {
        const h3 = wrapper.find('h3')
        expect( h3.text().trim()).toBe( title );
    })
    
    test('debe de tnere la imagen igual al url y alt al props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
    })
    test('debe tener animate__fadeIn', () => {
        const div =wrapper.find('div')
        const className = "animate__fadeIn"
        //console.log(div.prop("className"))
        expect(div.prop("className").includes(className) ).toBe(true);  
        // expect(div.prop("className")).toContain(className);
    })
    
    
})