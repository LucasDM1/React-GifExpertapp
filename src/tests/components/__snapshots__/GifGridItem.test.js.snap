// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGridItem /> Debe mostrar GifGridItem correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <h3>
    Un titulo
  </h3>
</div>
`;
